File: HCS.Core/Domain/ProvidedUtility.cs
using System.Collections.Generic;
using System.Collections.ObjectModel;

namespace HCS.Core.Domain
{
    public class ProvidedUtility
    {
        public int Id { get; set; }
        public int ProviderId { get; set; }
        public Provider Provider { get; set; }
        public int UtilityId { get; set; }
        public Utility Utility { get; set; }
        public ICollection<Tariff> Tariffs { get; set; }

        public ProvidedUtility()
        {
            Tariffs = new Collection<Tariff>();
        }
    }
}

File: HCS.Client/ClientApp/app/models/provided-utility.ts
import { TariffInfo } from "./tariff";

export class ProvidedUtility {
    id: number;
    utilityId: number;
    name: string;
    measureUnit: string;
    isSeasonal: boolean;
    tariffs: TariffInfo[];

    constructor() {
        this.tariffs = [];
    }
}